modules/shared/src/main/scala/io/constellationnetwork/env/JarSignature.scala

```diff
@@ -0,0 +1,20 @@
+package io.constellationnetwork.env
+
+import java.io.File
+import java.nio.file.Files
+
+import io.constellationnetwork.security.hash.Hash
+
+object JarSignature {
+
+  lazy val hash: Hash = jarHash
+
+  private def jarHash = {
+    val jarFile = new File(JarSignature.getClass.getProtectionDomain.getCodeSource.getLocation.toURI)
+    if (jarFile.isFile) {
+      val buffer = Files.readAllBytes(jarFile.toPath)
```

Comment:
The fat jar is usually around 100 MB, so this may be a bit resource intensive but our Hash assumes small preimage inputs so that would probably need to be refactored if we wanted to use it here. The other option may just be to reproduce some code and just use the bouncycastle `sha256` directly.

Another concern I had was about the reproducibility of this hash. I don't recall specifics but there may be an issue with far jars being slightly different builds between machines / environments. I believe the solution involved the `sbt reproducible builds` [plugin ](https://github.com/raboof/sbt-reproducible-builds) but the use case may have been slightly different.

Reply:
I agree with @scasplte2, it's a resource intensive operation, so it should be wrapped into an effect type that can efficiently handle cpu-intensive operations. @gclaramunt we do prefer functional style in the codebase (cats, cats-effect and fs2). Please take a look how it works and how it handles file i/o. reading bytes from file should be implemented using FS2, so in a streaming fashion, instead of single blocking operation.

Reply:
true, it can create a memory spike (specially if the jar keeps growing), and yeah, I found some scenarios where the hash is not consistent, I'm changing my approach.